src/Storages/StorageMerge.cpp

```diff
@@ -965,8 +1138,14 @@ void ReadFromMerge::convertingSourceStream(
     const Aliases & aliases,
     ContextPtr local_context,
     QueryPipelineBuilder & builder,
-    const QueryProcessingStage::Enum & processed_stage)
+    QueryProcessingStage::Enum processed_stage,
+    RowPolicyData & row_policy_data)
 {
+    if (row_policy_data.hasRowPolicy())
+    {
+        row_policy_data.addFilterTransform(builder);
+    }
```

Comment:
`convertingSourceStream()` is a function for converting the types of columns. It seems to me it would be better to call `row_policy_data.addFilterTransform()` directly from the caller function, i.e. from `ReadFromMerge::createSources()`